app/routines/newflow/update_salesforce_lead.rb

```diff
@@ -0,0 +1,59 @@
+module Newflow
+  class UpdateSalesforceLead
+    lev_routine
+
+    protected #################
+
+    def exec(user:)
+      status.set_job_name(self.class.name)
+      status.set_job_args(user: user.to_global_id.to_s)
+
+      lead_id = user.salesforce_lead_id
+      lead = fetch_lead(lead_id)
+
+      if not lead.present?
+        nonfatal_error(code: :lead_missing_in_salesforce)
+        log_error(user, lead)
+      elsif update_salesforce_lead!(lead, user)
+        log_success(user, lead)
+      else
+        log_error(user, lead)
+      end
+
+      outputs.lead = lead
+    end
+
+    private #################
+
+    def fetch_lead(lead_id)
+      OpenStax::Salesforce::Remote::Lead.find(lead_id)
+    end
+
+    def update_salesforce_lead!(lead, user)
+      lead.update(
+        # sheerid_verification_status: 'verified' # TODO: future feature that must be enabled by SF
+        first_name: user.first_name,
+        last_name: user.last_name,
+        school: user.sheerid_reported_school || user.self_reported_school || 'unknown',
+        # role: user.role,
```

Comment:
intentional to leave these commented lines in?

Reply:
Fixed.